pkg/odo/util/validation/validatable.go

```diff
@@ -0,0 +1,13 @@
+package validation
+
+// Validatable represents a common ancestor for validatable parameters
+type Validatable struct {
+	Required             bool        `json:"required,omitempty"`
+	Type                 string      `json:"type"`
+	AdditionalValidators []Validator `json:"-"`
+}
+
+// AsValidatable allows avoiding type casts in client code
```

Comment:
Can you explain how this can be helpful?
As in, at the moment I see the func returning the object its invoked upon

Reply:
This just helps using Validatable in client code to avoid casting. See https://github.com/redhat-developer/odo/pull/1126/files#diff-6750ef04ff7fb19ec7f108ffcc9df23aR298: without this function, we would need a cast on the call site and that would be a lot less elegant :)